File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/NotificationBase.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types.Notification;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public abstract class NotificationBase
    {
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;
        public int? Id { get; set; }
        public Guid? Uuid { get; set; }
        public NotificationStatusType? Status { get; set; } = NotificationStatusType.UNREAD;
        public string? Message { get; set; }
        public NotificationType Type { get; set;}
        public required List<NotificationRecipient> Recipients { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/notification/notification-base.ts
import { NotificationStatusType } from "../types/notification-status.types";
import { NotificationType } from "../types/notification.types";

export class NotificationBase {
  constructor(
    public createdAt: Date,
    public uuid: string,
    public message: string,
    public type: NotificationType,
    public status: NotificationStatusType
  ) { }
}


